main.ts

```diff
@@ -245,6 +237,7 @@ class PhysicalObject extends Entity {
     public mass: number;
     public moment_of_inertia: number;
     public center_of_mass: Vector2D;
+    public is_ground: boolean;
```

Comment:
This is much better.
I'd argue maybe this is a `GameElement`'s attribute not a `PhysicalObject`'s?

Reply:
Same thing for all `GameSet` references. Is it possible to get rid of all those in the collide methods, now that we have `update_game_set` in `GameElement`?